### PR TITLE
Separate metadata processor

### PR DESCRIPTION
**Pull request by MateoRojas**
Separates metadata into its own processor. The separation is meant to be done incrementally, right now it was implemented with the fewest code changes possible. In a future commit we will be refactoring the code so that Metadata and Run processors do not share the same DAO and CommandProcessor classes.

**Note:** Search is not working. Everything else is, at least everything that is covered by e2e tests.

### CODE REVIEW

File: server/src/main/java/io/littlehorse/common/LHConfig.java [hunk 1]
```diff
@@ -275,6 +283,17 @@ public static List<NewTopic> getAllTopics(
                 .configs(globalMetaCLConfig)
         );
 
+        // TODO: define the appropriate configuration for this topic -> For sure it shouldn't have compaction
```
Comment (MateoRojas) [inline server/src/main/java/io/littlehorse/common/LHConfig.java:286]: Is this the correct configuration for the metadata topic? 
@coltmcnealy-lh

Comment (coltmcnealy-lh) [reply server/src/main/java/io/littlehorse/common/LHConfig.java:286]: This is correct for the Metadata Command Topic.

File: server/src/main/java/io/littlehorse/server/streamsimpl/ServerTopology.java [hunk 2]
```diff
@@ -67,45 +69,88 @@ public class ServerTopology {
...
+        topo.addSource(
+            METADATA_SOURCE, // source name
+            Serdes.String().deserializer(), // key deserializer
+            new LHDeserializer<>(Command.class, config), // value deserializer
+            config.getMetadataCmdTopicName() // source topic
+        );
+
+        topo.addProcessor(
+            METADATA_PROCESSOR,
+            () ->
```
Comment (coltmcnealy-lh) [inline server/src/main/java/io/littlehorse/server/streamsimpl/ServerTopology.java:123]: I am regretting the decision to use `prettier`

File: server/src/main/java/io/littlehorse/server/streamsimpl/coreprocessors/CommandProcessor.java [hunk 3]
```diff
@@ -28,22 +29,42 @@ public class CommandProcessor
...
+        );
+        dao =
+            new KafkaStreamsLHDAOImpl(
+                this.ctx,
+                config,
+                server,
+                wfSpecCache,
+                localStore,
+                isMetadataProcessorInstance
+            );
```
Comment (coltmcnealy-lh) [inline server/src/main/java/io/littlehorse/server/streamsimpl/coreprocessors/CommandProcessor.java:67]: This is fine for now; however, in a follow-up PR let's add:
* `GlobalMetadataDAO`
* `ROGlobalMetaStore`

That should simplify the `KafkaStreamsLHDAOImpl`

**Review by coltmcnealy-lh — APPROVED** [review]
Good work. Thanks!

File: server/src/main/java/io/littlehorse/server/streamsimpl/ServerTopology.java (hunk 2 above)
Comment (MateoRojas) [reply server/src/main/java/io/littlehorse/server/streamsimpl/ServerTopology.java:123]: Hahahaha I was thinking the same. I think we need to configure it better :)

**Pull request merged**